File: Homework2_Derezovskiy/Homework2_Derezovskiy/parallelepiped.cpp
//------------------------------------------------------------------------------
// parallelepiped_In.cpp - содержит процедуру ввода параметров
// для уже созданного параллелепипеда
//------------------------------------------------------------------------------

#include "parallelepiped.h"

//------------------------------------------------------------------------------
// Ввод параметров параллелепипеда из файла
void Parallelepiped::In(std::istream& ifst) {
    ifst >> x >> y >> z >> materialDensity;
}

// Случайный ввод параметров параллелепипеда
void Parallelepiped::InRnd() {
    x = Shape::rnd20.Get();
    y = Shape::rnd20.Get();
    z = Shape::rnd20.Get();
    materialDensity = Shape::rnd20.Get();
}

//------------------------------------------------------------------------------
// Вывод параметров параллелепипеда в форматируемый поток
void Parallelepiped::Out(std::ostream& ofst) {
    ofst << "It is Parallelepiped: x = " << x << ", y = " << y
         << ", z = " << z << ". Material density = " << materialDensity
         << ". Square = " << Square() << "\n";
}

//------------------------------------------------------------------------------
// Вычисление периметра поверхности параллелепипеда
double Parallelepiped::Square() {
    return 2.0 * (x * y + y * z + x * z);
}

File: Homework3_Derezovskiy/src/parallelepiped.py
import random

class Parallelepiped:
    def __init__(self):
        self.x = 0
        self.y = 0
        self.z = 0
        self.materialDensity = 0

    def In(self, strArray):
        self.x = int(strArray[1])
        self.y = int(strArray[2])
        self.z = int(strArray[3])
        self.materialDensity = int(strArray[4])

    def InRnd(self):
        self.x = random.randint(1, 20)
        self.y = random.randint(1, 20)
        self.z = random.randint(1, 20)
        self.materialDensity = random.randint(1, 20)

    def Write(self, ostream):
        ostream.write("Parallelepiped: x = {}, y = {}, z = {}, materialDensity = {}, Square = {}".
                      format(self.x, self.y, self.z, self.materialDensity, self.Square()))
        pass

    def Square(self):
        return 2.0 * (self.x * self.y + self.y * self.z + self.x * self.z);
        pass
